packages/web-components/src/components/va-table/va-table.tsx

```diff
@@ -248,8 +248,22 @@ export class VaTable {
     // only do sort if sortable data in column
     if (compareFunc !== null) {
       rows.sort((a: Element, b: Element) => {
-        const _a = a.children[index].innerHTML.trim();
-        const _b = b.children[index].innerHTML.trim();
+        const cellA = a.children[index];
+        const cellB = b.children[index];
+        
+        // Check for data-sort-value attribute first
+        const sortValueA = cellA.getAttribute('data-sort-value');
+        const sortValueB = cellB.getAttribute('data-sort-value');
+        
+        // If both cells have data-sort-value attributes, use those
+        if (sortValueA !== null && sortValueB !== null) {
+          return compareFunc(sortValueA, sortValueB);
+        }
+        
+        // If only one cell has data-sort-value, or neither has it, fall back to innerHTML
```

Comment:
This should be able to handle three scenarios:

- Both compared cells have `data-sort-value`: Uses both attribute values for comparison
- Only one cell has `data-sort-value`: Uses the attribute value for that cell and falls back to innerHTML for the other
- Neither cell has data-sort-value: Falls back to innerHTML for both cells:  Falls back to innerHTML for both cells

Mixed data comparisons will still work (text vs. numeric attribute) but ideally it should be consistent to avoid unexpected sorting behavior. We will leave it to teams to implement in the way that makes best sense for their data though.